Add CrowdStrike Falcon sensor

jsf9k opened the pull request:
## 🗣 Description ##

This pull request:
- Adds the CrowdStrike Falcon sensor to the AMI.
- Gets rid of some odds and ends left over from when we used to include Tanium and Tenable in this AMI.

See also cisagov/ansible-role-crowdstrike#5.

## 💭 Motivation and context ##

- We must install the CrowdStrike Falcon sensor to maintain the COOL ATO.
- Leftover code should be removed, since it's unnecessary code.  Deleted code is code that cannot contain bugs.

## 🧪 Testing ##

All automated tests pass.  I also built a new OpenVPN AMI with these changes for our staging COOL environment and verified that:
- OpenVPN still functions.
- The CrowdStrike Falcon sensor has the correct customer ID and tags and starts up successfully.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.
- [x] Build and test a staging AMI that contains the changes from this PR.

## ✅ Pre-merge checklist ##

- [x] Revert to the default branch of [cisagov/ansible-role-crowdstrike](https://github.com/cisagov/ansible-role-crowdstrike) once cisagov/ansible-role-crowdstrike#5 is merged.
- [x] Revert to the default branch of [cisagov/ansible-role-openvpn](https://github.com/cisagov/ansible-role-openvpn) once cisagov/ansible-role-openvpn#73 is merged.
- [x] Finalize version.

## ✅ Post-merge checklist ##

- [x] Create a release.

jsf9k committed:
Bump version from 0.3.3 to 0.3.4

jsf9k committed:
Bump version from 0.3.4 to 0.3.4-rc.1

jsf9k committed:
Add code to install CrowdStrike Falcon sensor

Note that we are temporarily using a non-default branch of
cisagov/ansible-role-crowdstrike.  We can use the default branch once
cisagov/ansible-role-crowdstrike#5 is merged.

jsf9k committed:
Update TF code for build user

* Remove some old cruft that is no longer necessary since we removed
the Nessus and Tanium agents
* Add some new code to allow the build user to get the CrowdStrike
Falcon sensor package from the third-party S3 bucket.

jsf9k committed:
Temporarily use a non-default branch of cisagov/ansible-role-openvpn

This is being done temporarily for testing, and this change can be
reverted once cisagov/ansible-role-openvpn#73 is merged.

jsf9k committed:
Bump version from 0.3.4-rc.1 to 0.3.4-rc.2

mcdonnnj commented on src/cdm.yml:
```diff
@@ -0,0 +1,26 @@
+---
+- hosts: all
+  name: Configure for the CISA CDM environment
+  vars:
+    ports:
+      # CrowdStrike
+      - direction: out
+        port: 443
+        proto: tcp
```
Did we want to reposition this to be more alphabetically sorted?

jsf9k replied:
Please see commit 4ccac31.

jsf9k replied:
Please see commit 4ccac31.

mcdonnnj reviewed (approved):
Seems straightforward enough. One comment for your consideration.

jsf9k committed:
Move vars to the task where they are to be applied

It is better to scope the vars as specifically as possible.

Co-authored-by: Nicholas McDonnell <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Bump version from 0.3.4-rc.2 to 0.3.4-rc.3

dav3r reviewed (approved):
It looks like this PR also removes some leftover bits related to CDM Tanium and Nessus.  Can you please mention that in the PR description?

jsf9k commented:
> It looks like this PR also removes some leftover bits related to CDM Tanium and Nessus. Can you please mention that in the PR description?

Good call.  Description updated.

jsf9k committed:
Revert to the default branch of cisagov/ansible-role-openvpn

We can do this now that cisagov/ansible-role-openvpn#73 has been
merged.

jsf9k committed:
Remove CDM certificate dependencies

CrowdStrike Falcon sensor only reaches out to a CrowdStrike-owned web
server, and does not require these certificates or make use of the CDM
IPSec tunnel.

jsf9k committed:
Bump version from 0.3.4-rc.3 to 0.3.4-rc.4

dav3r reviewed (approved):
👍

mcdonnnj reviewed (approved):
LGTM ✔

jsf9k commented:
> LGTM heavy_check_mark

:open_mouth:

jsf9k committed:
Revert to using the default branch of cisagov/ansible-role-crowdstrike

We can do this now that cisagov/ansible-role-crowdstrike#5 has been
merged.

jsf9k committed:
Finalize version from 0.3.4-rc.4 to 0.3.4
